src/otx/core/data/transform_libs/torchvision.py

```diff
@@ -3119,11 +3124,100 @@ def generate(cls, config: SubsetConfig) -> Compose:
 
         transforms = []
         for cfg_transform in config.transforms:
+            if (input_size := getattr(config, "input_size", None)) is not None:
+                cls._eval_input_size(cfg_transform, input_size)
             transform = cls._dispatch_transform(cfg_transform)
             transforms.append(transform)
 
         return Compose(transforms)
 
+    @classmethod
+    def _eval_input_size(cls, cfg_transform: dict[str, Any], input_size: int | tuple[int, int] | None) -> None:
+        """Evaluate the input_size and replace the placeholder in the init_args.
+
+        Input size should be specified as $(input_size). (e.g. $(input_size) * 0.5)
+        Only simple multiplication or division evaluation is supported. For example,
+        $(input_size) * -0.5    => supported
+        $(input_size) * 2.1 / 3 => supported
+        $(input_size) + 1       => not supported
+        The function decides to pass tuple type or int type based on the type hint of the argument.
+        float point values are rounded to int.
+        """
+        if input_size is None:
+            return
+        _input_size: tuple[int, int] = (input_size, input_size) if isinstance(input_size, int) else tuple(input_size)  # type: ignore[assignment]
```

Comment:
Do we need to use this conversion from int to tuple?
I'm not sure if it is necessary because L3170 uses a single integer.

Reply:
If we didn't change it to tuple, then we should check what type `input_size` is currently because `input_size` can be either int or tuple. So, we need to add a few code to check current type of `input_size` after `if available_types is None or check_type(_input_size, available_types)`. For this reason, I think fixing input_size type to tuple makes code more readable.

Reply:
I don't think about it.
Please check the below example.
```python
model_cls = get_obj_from_str("otx.core.data.transform_libs.torchvision.Resize") # <class 'otx.core.data.transform_libs.torchvision.Resize'>
key = "scale"
val = "$(input_size)"
available_types = typing.get_type_hints(model_cls.__init__).get(key) # int | tuple[int, int] | None

# for tuple
_input_size = (1, 1)
check_type(_input_size, available_types) # True with (1, 1)

# for int
_input_size = 1
check_type(_input_size, available_types) # True with 1

# for str
_input_size = "1"
check_type(_input_size, available_types) # False with TypeCheckError at L3153

# both tuple and int can use the same argument for `cls._safe_eval`
cls._safe_eval(val.replace("$(input_size)", str(_input_size))
```
I think we don't need an additional condition you said because `check_type` can support both types.
If I'm wrong, please correct me :)

Reply:
I thought it again, you seem to consider the cases about only single tuple argument or int argument, is it correct?
If so, I think your choice is more safe, please ignore it and let's keep this.